### PR TITLE
Fix nil pointer in etcd tools

### LINKED ISSUE #6934: kube-apiserver panic information in logs when doing a list with fieldSelector that single matches
With the move to running the kube-apiserver in a pod, I am getting errors of the form:
```
I0416 21:41:19.321622       1 handlers.go:137] GET /api/v1beta3/nodes: (2.578173ms) 200 [[kube-controller-manager/v0.15.0 (linux/amd64) kubernetes/fa4363d] 127.0.0.1:44572]
E0416 21:41:20.039309       1 master.go:357] recover from panic situation: - runtime error: invalid memory address or nil pointer dereference
    /usr/src/go/src/runtime/asm_amd64.s:401
    /usr/src/go/src/runtime/panic.go:387
    /usr/src/go/src/runtime/panic.go:42
    /usr/src/go/src/runtime/sigpanic_unix.go:26
    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/tools/etcd_helper.go:178
    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd/etcd.go:153
    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd/etcd.go:142
    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/resthandler.go:177
    /go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/container.go:214
    /go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/filter.go:21
    /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/apiserver.go:84
    /go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/filter.go:19
    /go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/container.go:216
    /go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/emicklei/go-restful/container.go:90
    /usr/src/go/src/net/http/server.go:1265
```
The root null pointer is from this line:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/tools/etcd_helper.go#L168

The line numbers in the stack are from my continuing debug of the original null pointer.

We need to protect against a null pointer in this path, which appears to only be invoked when doing a list call that matches a single item based on field selector.

Will submit a PR to address.

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Fixes https://github.com/GoogleCloudPlatform/kubernetes/issues/6934

/cc @smarterclayton

### CODE REVIEW

File: pkg/tools/etcd_helper.go [hunk 1]
```diff
@@ -164,6 +164,10 @@ func (h *EtcdHelper) ExtractObjToList(key string, listObj runtime.Object) error
 		return err
 	}
 
+	if response == nil {
```
Comment (smarterclayton) [inline pkg/tools/etcd_helper.go]: Can you put this inside if err != nil, and have if IsEtcdNotFound(err) return nil?  Would be clearer.

**Comment by roberthbailey** [comment]
I ran across this in #6949 and fixed it differently. I'm happy to take this change instead of mine if this way makes more sense.

**Comment by smarterclayton** [comment]
The fix i asked Derek for is slightly different and is a bit more consistent with other code. Don't care which pull fixes it though.

**Comment by roberthbailey** [comment]
SGTM. I'll rebase once this PR is merged.

**Comment by derekwaynecarr** [comment]
@smarterclayton merge it ;-)

**Comment by roberthbailey** [comment]
+1

**Pull request merged**